typescript/cli/src/deploy/warp.ts

```diff
@@ -457,94 +448,119 @@ export async function runWarpRouteApply(
-      objMap(warpDeployConfig, async (chain, config) => {
-        try {
-          config.ismFactoryAddresses = addresses[
-            chain
-          ] as ProxyFactoryFactoriesAddresses;
-          const evmERC20WarpModule = new EvmERC20WarpModule(
-            multiProvider,
-            {
-              config,
-              chain,
-              addresses: {
-                deployedTokenRoute:
-                  warpCoreConfigByChain[chain].addressOrDenom!,
-              },
-            },
-            contractVerifier,
-          );
-          const transactions = await evmERC20WarpModule.update(config);
-
-          if (transactions.length == 0)
-            return logGreen(
-              `Warp config on ${chain} is the same as target. No updates needed.`,
-            );
-          await submitWarpApplyTransactions(chain, params, transactions);
-        } catch (e) {
-          logRed(`Warp config on ${chain} failed to update.`, e);
-        }
-      }),
-    );
-  } else if (warpDeployChains.length > warpCoreChains.length) {
-    logGray('Extending deployed Warp configs');
 
-    // Split between the existing and additional config
-    const existingConfigs: WarpRouteDeployConfig = objFilter(
-      warpDeployConfig,
-      (chain, _config): _config is any => warpCoreChains.includes(chain),
-    );
+  // Split between the existing and additional config
+  const existingConfigs: WarpRouteDeployConfig = objFilter(
+    warpDeployConfig,
+    (chain, _config): _config is any => warpCoreChains.includes(chain),
+  );
 
-    let extendedConfigs: WarpRouteDeployConfig = objFilter(
-      warpDeployConfig,
-      (chain, _config): _config is any => !warpCoreChains.includes(chain),
-    );
+  let extendedConfigs: WarpRouteDeployConfig = objFilter(
+    warpDeployConfig,
+    (chain, _config): _config is any => !warpCoreChains.includes(chain),
+  );
 
-    extendedConfigs = await deriveMetadataFromExisting(
-      multiProvider,
-      existingConfigs,
-      extendedConfigs,
-    );
+  if (isObjEmpty(extendedConfigs)) return [];
 
-    const newDeployedContracts = await executeDeploy(
-      {
-        // TODO: use EvmERC20WarpModule when it's ready
-        context,
-        warpDeployConfig: extendedConfigs,
-      },
-      apiKeys,
-    );
+  extendedConfigs = await deriveMetadataFromExisting(
+    multiProvider,
+    existingConfigs,
+    extendedConfigs,
+  );
 
-    const mergedRouters = mergeAllRouters(
-      multiProvider,
-      existingConfigs,
-      newDeployedContracts,
-      warpCoreConfigByChain,
-    );
+  const newDeployedContracts = await executeDeploy(
+    {
+      // TODO: use EvmERC20WarpModule when it's ready
+      context: params.context,
+      warpDeployConfig: extendedConfigs,
+    },
+    apiKeys,
+  );
 
-    await enrollRemoteRouters(params, mergedRouters);
+  const mergedRouters = mergeAllRouters(
+    multiProvider,
+    existingConfigs,
+    newDeployedContracts,
+    warpCoreConfigByChain,
+  );
```

Comment:
can we just join the newDeployedContracts with the existing deployed and pass that to the update step?

Reply:
i can try to do that in a next PR. wanted to keep the diffs to a minimal for this